### PR TITLE
Add a new vitakinetic power to restore lost limbs

### PR DESCRIPTION
**Pull request by MNG-cataclysm**
#### Summary
Mods "Add a new vitakinetic power to restore lost limbs."

#### Purpose of change
I thought it would be pretty neat if we had a power to heal limbs, now that there's a few different base-game ways to do that. When I checked the documentation for new powers, this sort of ability was explicitly mentioned under the vitakinetic section, so I decided to go ahead with it.

#### Describe the solution
Add a new vitakinetic power: Ameliorate Extremity. This top-tier ability is one of the hardest spells a vitakinetic can learn, but it offers the power to restore lost limbs. However, this has some drawbacks. These are as follows:

1. It takes over eight hours to successfully use this power.
2. It is extremely draining, adding a lot of psychic wear to the caster and having a 100% chance to become drained afterwards.
3. It requires a strong concentration to use, meaning that pain and bothering things make it exponentially harder to use.
4. It takes a lot of bodily power to use, about 2.5 times as much as Anabolic Rejuvenation.

Additionally, it is also difficult to learn, being taught by Anabolic Rejuvenation at spell level 10. Contemplating it also takes twice as much green crystalline dust as the other powers, but I'd say this is worth it.

#### Describe alternatives you've considered
Not adding this power at all or lowering the requirements of it.

#### Testing
Went into the game and gave myself the power. My para-amputee was no longer an amputee afterwards.

#### Additional context
I would like for this power to be able to be used on allies, but I'd have to retool the EOCs and whatnot to allow for this. Also, I don't know if that would violate the lore of how vitakinesis works for Mind Over Matter, @Standing-Storm.
I also know that you cannot gain any of the lost limb traits in-game at the moment, but I'm planning to model limb loss at a later date, so I thought I'd go ahead and do this now so it's ready to be used. 
I decided to add all of these drawbacks to the spell as a way to make it balanced, as I figured it would be quite difficult to regrow a whole limb from scratch. This would also prevent the spell from being spammed and making losing a limb a trivial issue.

When missing arms are added to the game, I'll come back and make this spell able to heal those, too.

### CODE REVIEW

**Comment by Maleclypse** [comment]
Repost I’m ready to merge after Standing Storm approves. I don’t want to go hog wild if this needs any changes per them.

**Comment by Cupidatis** [comment]
Is it working with broken limbs?

**Comment by MNG-cataclysm** [comment]
> Is it working with broken limbs? 

No, it interfaces with missing limbs and their traits. Right now, this only really works with certain start professions because you can't lose limbs in game yet, but that should change soon.

**Comment by Standing-Storm** [comment]
Alright, now I've had a chance to take a look. Thank you for doing this, especially since you're the person working on missing limbs so you have a better understanding of those mechanics. I'll follow this up with specific suggestions, but as a general comment: 

> I would like for this power to be able to be used on allies, but I'd have to retool the EOCs and whatnot to allow for this. Also, I don't know if that would violate the lore of how vitakinesis works for Mind Over Matter,

My plan for vitakinesis is that it mostly only works on the actual psion, and all of the powerful effects only work on the psion--that's why you can't directly cure diseases in others, you have to absorb it (_The Green Mile_ style) and then cure it in yourself. Absorbing a missing limb would be much more like magic than psionics.

_But_ the line about matrix technology being able to overcome these limitations is in there for situations like this--you could have a XEDRA machine that allows the psion to use Ameliorate Extremity on others, maybe requiring a charged vitakinetic matrix crystal for an operator who doesn't personally know Ameliorate Extremity but a drained one for a psion who does. I already had the idea for a biokinetic/vitakinetic Project PHAVIAN research location under a hospital that would be a perfect place to put it.

File: data/mods/MindOverMatter/powers/vitakinesis_eoc.json
Comment (Standing-Storm) [inline data/mods/MindOverMatter/powers/vitakinesis_eoc.json:551]:
```suggestion
    "//":  "About 2.5 times as costly as the super heal drain.  You are regrowing a whole body part, you know.",
```
The way that Anabolic Rejuvenation works, it runs that drain every 30 seconds when a tick of healing occurs, whereas Ameliorate Extremity only runs the drain once. If it's intended to be 2.5 times at costly it would have to be much higher, since level 1 Anabolic Rejuvenation will tick an average of 8 times, causing the calorie/vitamin/etc drain each time.

File: data/mods/MindOverMatter/recipes/practice/vitakinesis_practice.json
Comment (Standing-Storm) [inline data/mods/MindOverMatter/recipes/practice/vitakinesis_practice.json:641]: I think the double dust cost is a good (and should probably be expanded to other level 10 powers), but the time should be the same as the other practice recipes at 30 minutes. It's all balanced around how much XP reading spellbooks gives in Magiclysm and that provides the same XP-per-second regardless of Difficulty.

File: data/mods/MindOverMatter/powers/vitakinesis_eoc.json
Comment (Standing-Storm) [inline data/mods/MindOverMatter/powers/vitakinesis_eoc.json:499]:
```suggestion
      { "u_message": "Your right leg has been restored.", "type": "good" },
```

File: data/mods/MindOverMatter/powers/vitakinesis_eoc.json
Comment (Standing-Storm) [inline data/mods/MindOverMatter/powers/vitakinesis_eoc.json:536]:
```suggestion
      { "u_message": "Your left leg has been restored.", "type": "good" },
```

File: data/mods/MindOverMatter/powers/vitakinesis.json
Comment (Standing-Storm) [inline data/mods/MindOverMatter/powers/vitakinesis.json:494]:
```suggestion
    "difficulty": 10,
```
Based on the practice difficulty, I assume this is meant to be a Difficulty 10 power.

File: data/mods/MindOverMatter/powers/vitakinesis_eoc.json
Comment (MNG-cataclysm) [reply data/mods/MindOverMatter/powers/vitakinesis_eoc.json:551]: Does Anabolic Rejuvenation's drain tick more than 8 times at higher levels?

Comment (Standing-Storm) [reply data/mods/MindOverMatter/powers/vitakinesis_eoc.json:551]: Yes. It and the healing are set to run once every 30 seconds, so it becomes more powerful by increasing duration, which increases the total healing done and the total calorie/vitamin cost. It's 30-60 seconds (random) increased duration per level, so 1-2 ticks.

Comment (MNG-cataclysm) [reply data/mods/MindOverMatter/powers/vitakinesis_eoc.json]: Alright. I ran some math on it and have increased the scaling using a level 12 spell approximation.

**Review by Standing-Storm — APPROVED** [review]

**Pull request merged**
